recipes/boost/all/conanfile.py

```diff
@@ -307,7 +307,7 @@ def _shared(self):
 
     @property
     def _stacktrace_addr2line_available(self):
-        return not self.options.header_only and not self.options.without_stacktrace and self.settings.compiler != "Visual Studio"
+        return not self.options.header_only and not self.options.without_stacktrace and self.settings.os != "Windows"
```

Comment:
It's not sufficient for MinGW. There is additional logic in `package_info()` when available librarires are checked:

```python
            def filter_transform_module_libraries(names):
                libs = []
                for name in names:
                    if name in ("boost_stacktrace_windbg", "boost_stacktrace_windbg_cached") and self.settings.os != "Windows":
                        continue
                    if name in ("boost_stacktrace_addr2line", "boost_stacktrace_backtrace", "boost_stacktrace_basic",) and self.settings.compiler == "Visual Studio":
                        continue
                    if not self.options.get_safe("numa") and "_numa" in name:
                        continue
                    libs.append(add_libprefix(name.format(**libformatdata)) + libsuffix)
                return libs
```

So now `boost_stacktrace_addr2line` should be filtered if Windows. But actually it fails anyway with MinGW, because `boost_stacktrace_backtrace` & `boost_stacktrace_basic` are not packaged (or built).

Here is the command with all components disabled except stacktrace:

```
conan create . boost/1.76.0@ -o boost:without_atomic=True -o boost:without_chrono=True -o boost:without_container=True -o boost:without_context=True -o boost:without_contract=True -o boost:without_coroutine=True -o boost:without_date_time=True -o boost:without_exception=True -o boost:without_fiber=True -o boost:without_filesystem=True -o boost:without_graph=True -o boost:without_iostreams=True -o boost:without_json=True -o boost:without_locale=True -o boost:without_log=True -o boost:without_math=True -o boost:without_nowide=True -o boost:without_program_options=True -o boost:without_random=True -o boost:without_regex=True -o boost:without_serialization=True -o boost:without_stacktrace=False -o boost:without_system=True -o boost:without_test=True -o boost:without_thread=True -o boost:without_timer=True -o boost:without_type_erasure=True -o boost:without_wave=True
```

Reply:
I've dropped boost_stacktrace_addr2line, boost_stacktrace_backtrace and boost_stacktrace_basic on Windows, not only Visual Studio. I've tried with MinGW, and these libs are not built, whatever I try to do.

Reply:
Is this mingw@Windows or mingw@Linux?
Because I have not been able to build boost for mingw on Windows.

Reply:
MinGW Windows

Reply:
Actually, build fails with MinGW and default options, but I've found the component to disable, so that I can test boost recipe with MinGW: https://github.com/conan-io/conan-center-index/issues/4764#issuecomment-845807588